File: Leetcode/Q121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    if(prices.length < 2)
        return 0
    
    var minp = prices[0]
    var maxp = prices[1]-prices[0]
    
    for(let i=1; i<prices.length; i++){
        var possible_profit = prices[i]-minp
        maxp = Math.max(maxp, possible_profit)
        minp = Math.min(minp, prices[i])
    }
    return (maxp>0? maxp: 0)
};

File: Leetcode/Q121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices:
            return 0
        
        maxP = 0
        minP = prices[0]
        
        for i in range(1, len(prices)):
            if prices[i] < minP:
                minP = prices[i]
            tempMax = prices[i] - minP
            if(tempMax > maxP):
                maxP = tempMax
                
        return maxP
            
